### PR TITLE
fix(rust): type untyped rpc::parse_response() calls

### PR DESCRIPTION
**Pull request by neil2468**
## Current Behavior

The `ockam vault create` and `ockam identity create` commands fail to parse responses from `ockam_api`.

For example... 

```
> ockam node create n1 --skip-defaults
> ockam vault create --node n1
Failed to decode response body: end of input bytes
```

```
> ockam node create n1 --skip-defaults
> ockam identity create --node n1
Failed to decode response body: unexpected type map at position 18: expected array
```

## Proposed Changes

* Type the untyped uses of `Rpc::parse_response()` so that `ockam_command` correctly parse responses from `ockam_api`.
* Added tests in `commands.bats`.

Now (inc. check that we can't create multiple vaults or identities)...

```
> ockam node create n1 --skip-defaults
> ockam vault create --node n1
Vault created!
> ockam vault create --node n1
An error occurred while processing the request. Status code: 500 InternalServerError. Message: failed to handle request: Vault already exists
```

```
> ockam node create n1 --skip-defaults
> ockam identity create --node n1
Identity created!
> ockam identity create --node n1
An error occurred while processing the request. Status code: 500 InternalServerError. Message: failed to handle request: Identity already exists
```

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

### CODE REVIEW

File: implementations/rust/ockam/ockam_command/src/identity/create.rs [hunk 1]
```diff
@@ -26,7 +27,7 @@ async fn run_impl(
     let mut rpc = Rpc::background(&ctx, &options, &cmd.node_opts.api_node)?;
     let request = Request::post("/node/identity");
     rpc.request(request).await?;
-    rpc.parse_response()?;
+    rpc.parse_response::<CreateIdentityResponse>()?;
 
     println!("Identity created!");
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/src/identity/create.rs]: In a previous PR, we lost some information here. It should be:

```rust
res = rpc.parse_response::<CreateIdentityResponse>()?;
println!("Identity {} created!", res.identity_id)
```

Could you add it back while we are here?

**Review by adrianbenavides — COMMENTED** [review]
Nice catch, we totally missed this! I left a small suggestion :pray:

File: implementations/rust/ockam/ockam_command/src/identity/create.rs (hunk 1 above)
Comment (neil2468) [reply implementations/rust/ockam/ockam_command/src/identity/create.rs]: Done. Thanks.

**Comment by neil2468** [comment]
FYI: After rebasing there was an issue with `ockam vault create` not working. It complained that `-v` was the option for `verbose` __and__ `vault_name`. So, I've removed the short (`-v`) option for `vault_name`. Seems there are no tests for the new `vault_name` functionality.

**Comment by adrianbenavides** [comment]
> FYI: After rebasing there was an issue with `ockam vault create` not working. It complained that `-v` was the option for `verbose` **and** `vault_name`. So, I've removed the short (`-v`) option for `vault_name`. Seems there are no tests for the new `vault_name` functionality.

Thanks for reporting this, I'll create a new issue to add tests for that!

**Review by adrianbenavides — APPROVED** [review]
We are working towards increasing the test coverage of the commands and this PR was definitely very helpful. Thanks a lot @neil2468!

**Pull request merged**